### PR TITLE
Dbp 282 helm charts ci steps

### PR DESCRIPTION
**Pull request by maxi418**
# Description

## Approval for review
- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

File: .github/workflows/release-chart.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Push Charts to helm-charts-registry
+
+on:
+  push:
+    tags:
+      - 'dbildungs-iam-*'
```
Comment (aimee-889) [inline .github/workflows/release-chart.yml:6]: please limit the tags a bit more. see example https://github.com/hpi-schul-cloud/infra-tools/blob/master/.github/workflows/bapontag.yaml

File: .github/workflows/chart-release-manual.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  scan:
+    uses: dBildungsplattform/dbp-github-workflows/.github/workflows/check-helm-kics.yaml@1.3.0
+    permissions:
+      contents: read
+  release:
+    needs: scan
+    uses: dBildungsplattform/dbp-github-workflows/.github/workflows/chart-release.yaml@1.3.0
+    secrets: inherit
+    with:
+      chart_name: dbildungs-iam
```
Comment (aimee-889) [inline .github/workflows/chart-release-manual.yml:25]: think it should be dbildungscloud-iam-server

File: .github/workflows/chart-release-on-push-to-branch.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Push Charts to helm-charts-registry on push to branch
+
+on:
+  push:
+    branches-ignore:
+      - 'main'
+    paths:
+      - 'charts/**'
```
Comment (aimee-889) [inline .github/workflows/chart-release-on-push-to-branch.yml:8]: if changes are done that are only application wide, no helm chart gets published for the branch. In this case the pipeline would have to use a default (main) Helm Chart. This should be talked about in the Topic Team and taken into consideration

File: charts/dbildungs-iam/Chart.yaml [hunk 4]
```diff
@@ -1,11 +1,11 @@
 apiVersion: v2
 name: dbildungs-iam
```
Comment (aimee-889) [inline charts/dbildungs-iam/Chart.yaml:2]: i think the helm chart name and repo name should align and this should be corrected

File: .github/workflows/chart-release-manual.yml (hunk 2 above)
Comment (YannickEvers) [reply .github/workflows/chart-release-manual.yml:25]: Planned for later, probably together with moving the keycloak subchart into it's own repository in the next PI topic.

File: charts/dbildungs-iam/Chart.yaml (hunk 4 above)
Comment (YannickEvers) [reply charts/dbildungs-iam/Chart.yaml:2]: Planned for later, probably together with moving the keycloak subchart into it's own repository in the next PI topic.

**Review by aimee-889 — APPROVED** [review]

**Pull request merged**